### PR TITLE
Adds global cyborg mob list, refactors some silicon related loops

### PR DESCRIPTION
**Pull request by HowToLoLu**
## About The Pull Request

Does some code cleanup around Show-Borg-Panel: namely adding a list tracking all cyborgs (and implements this elsewhere in code), refactoring some for loops to be more cleanly written, and making it so that the verb no longer lets you select an invalid option.

## Why It's Good For The Game

Refactors code to be nicer, having a dedicated list for cyborgs seems reasonable if there's one for AI's, and reduces potential for user error during input for an admin tool. (Also makes it clear that AI's do not have a borg panel)

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

</details>

## Changelog
:cl:
fix: The "Show Borg Panel" admin verb no longer shows AI's, which are incompatible with the Borg Panel
code: Introduced a new global list for cyborgs; cyborg_list
refactor: Cleaned up some uses of silicon related mob lists
/:cl:

### CODE REVIEW

File: code/modules/admin/verbs/borgpanel.dm [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
 
-	if (!istype(borgo, /mob/living/silicon/robot))
-		borgo = input("Select a borg", "Select a borg", null, null) as null|anything in sort_names(GLOB.silicon_mobs)
-	if (!istype(borgo, /mob/living/silicon/robot))
-		to_chat(usr, "<span class='warning'>Borg is required for borgpanel</span>")
+	if(!length(GLOB.cyborg_list))
+		to_chat(usr, "<span class='warning'>There are no borgs to show a panel for!</span>")
+		return
+
+	if(!istype(borgo))
```
Comment (EvilDragonfiend) [inline code/modules/admin/verbs/borgpanel.dm]:
```suggestion
	if(isnull(borgo))
```

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]
basically removing shit codes and improving. easy and simple to approve

File: code/modules/admin/verbs/borgpanel.dm (hunk 1 above)
Comment (HowToLoLu) [reply code/modules/admin/verbs/borgpanel.dm]: ~~Don't you mean `!isnull(borgo)`?~~ (EDIT realized my mistake in interpretation a few seconds after replying) either way I'll have to resolve this in the morning, I've already shut down my PC and this app doesn't seem to allow me to apply changes at all.

File: code/modules/admin/verbs/borgpanel.dm [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
+		borgo = input("Select a borg", "Select a borg", null, null) as null|anything in sort_names(GLOB.cyborg_list)
+		if(!borgo)
+			return
+
+	if(QDELING(borgo))
```
Comment (EvilDragonfiend) [inline code/modules/admin/verbs/borgpanel.dm:19]: somewhere here needs `if(!is_cyborg(borgo))` here. QDELING is not enough.

Comment (HowToLoLu) [reply code/modules/admin/verbs/borgpanel.dm:19]: That's what the !istype check was supposed to be for

**Review by EvilDragonfiend — APPROVED** [review]
Looks good to me

**Pull request merged**